cli/cmd/diagnose.go

```diff
@@ -0,0 +1,363 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"github.com/odigos-io/odigos/cli/cmd/resources"
+	"github.com/odigos-io/odigos/cli/pkg/kube"
+	"github.com/odigos-io/odigos/k8sutils/pkg/client"
+	"github.com/spf13/cobra"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"sync"
+	"time"
+)
+
+const (
+	logBufferSize   = 1024 * 1024 // 1MB buffer size for reading logs in chunks
+	LogsDir         = "Logs"
+	CRDsDir         = "CRDs"
+	CRDName         = "crdName"
+	CRDGroup        = "crdGroup"
+	actionGroupName = "actions.odigos.io"
+	odigosGroupName = "odigos.io"
+)
+
+var (
+	diagnoseDirs = []string{LogsDir, CRDsDir}
+	CRDsList     = []map[string]string{
```

Comment:
Any Ideas for doing the Actions part dynamically?
Someone who will add a new action will forget to add it here.